### PR TITLE
Refactor/components

### PR DESCRIPTION
**Pull request by pepina-dev**
# Context
This project had repeated button styles across components and also some type warnings

# What has been done
The base button component is added and used in the other components of the project. Type issues were fixed. Also some feedback regarding to unnecessary clicks, and texts were applied.

#### Removed unneccesary double click
Before

https://github.com/user-attachments/assets/c1c44b91-8064-4f6c-b5b6-67f858df09cf

After

https://github.com/user-attachments/assets/9412fcc9-3f8b-4edf-9e75-b0ddc02cef74

#### Added button to mostro website
|Before |After|
|----|----|
|Screenshot 2024-12-20 at 12 34 32 PM|Screenshot 2024-12-20 at 11 16 48 AM|

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `<base-button>` component for improved button functionality and styling.

- **Updates**
	- Replaced traditional anchor elements with the `<base-button>` component in various banners for enhanced user interaction.
	- Enhanced the content and styling of the "About Us" banner for better presentation and readability.
	- Improved security by adding `rel="noopener noreferrer"` to all anchor tags in the "Contact" banner.

- **Configuration**
	- Added TypeScript support and type checking in the project configuration files.
	- Updated ESLint and TypeScript configurations to improve code quality and compatibility.

- **Dependencies**
	- Added new development dependencies for TypeScript support in the project.

### CODE REVIEW

File: components/base-button.vue [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  </a>
+  <a
+    v-else
+    :class="buttonStyles"
+    :href="href"
+  >
+    {{ label }}
+    <slot name="icon" />
+  </a>
+</template>
```
[the remark this answers is not in the document]
Comment (pepina-dev) [reply components/base-button.vue]: fixed!! ✅

File: components/contact-banner.vue [hunk 2]
```diff
@@ -5,15 +5,31 @@
...
     </h2>
     <div class="flex w-full justify-center gap-x-6 lg:gap-x-8">
-      <a target="_blank" href="https://x.com/MostroP2P" class="h-12 lg:h-16 rounded-full bg-secondary-400 hover:bg-secondary-300 shadow-lg flex aspect-square items-center justify-center">
+      <a
+        href="https://x.com/MostroP2P"
+        target="_blank"
+        rel="noopener noreferrer"
+        class="h-12 lg:h-16 rounded-full bg-secondary-400 hover:bg-secondary-300 shadow-lg flex aspect-square items-center justify-center"
+
+      >
```
[the remark this answers is not in the document]
Comment (pepina-dev) [reply components/contact-banner.vue]: ✅

File: components/base-button.vue [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+const props = withDefaults(defineProps<BaseButtonProps>(), {
+  scroll: false,
+});
+
+const buttonStyles = computed(() => {
+  return ([
+    "flex text-center items-center gap-2 font-semibold shadow-lg px-10 py-2 rounded-full transition-all duration-400 ease-in-out",
+    themeClasses[props.theme],
+  ]);
+});
```
[the remark this answers is not in the document]
Comment (pepina-dev) [reply components/base-button.vue]: ✅

File: components/base-button.vue [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+const handleClick = (event: Event) => {
+  if (props.scroll) {
+    event.preventDefault();
+    smoothScroll(props.href);
+  }
+};
```
[the remark this answers is not in the document]
Comment (pepina-dev) [reply components/base-button.vue:40]: I fixed it inside of the` smoothScroll` method

**Review by grunch — APPROVED** [review]
LGTM, I only have a comment, when you upload videos please use open standards like ogv

**Pull request merged**